code/modules/spells/spell_types/bloodcrawl.dm

```diff
@@ -14,6 +14,15 @@
 	action_background_icon_state = "bg_demon"
 	var/phased = FALSE
 
+/obj/effect/proc_holder/spell/bloodcrawl/cast_check(skipcharge = 0,mob/user = usr)
+	. = ..()
+	if(!.)
+		return FALSE
+	var/area/noteleport_check = get_area(user)
+	if(noteleport_check && noteleport_check.area_flags & NOTELEPORT)
+		to_chat(user, "<span class='danger'>Some dull, universal force is between you and your other existence, preventing you from blood crawling.</span>")
+		return FALSE
```

Comment:
I feel like you could compress these few lines of repeated code into a more generic helper function

Reply:
Honestly best solution would be combining all JAUNTS into one.

Reply:
yes but until then ;)